test/test_importer.py

```diff
@@ -1755,96 +1679,60 @@ def test_reimported_album_not_preserves_flexattr(self):
         self.assertEqual(self._album().data_source, "match_source")
 
 
-class ImportPretendTest(_common.TestCase, ImportHelper):
+class ImportPretendTest(ImportTestCase):
     """Test the pretend commandline option"""
 
-    def __init__(self, method_name="runTest"):
-        super().__init__(method_name)
-        self.matcher = None
-
     def setUp(self):
-        self.io = _common.DummyIO()
-        self.setup_beets()
-        self.__create_import_dir()
-        self.__create_empty_import_dir()
-        self._setup_import_session()
-        config["import"]["pretend"] = True
+        super().setUp()
         self.matcher = AutotagStub().install()
         self.io.install()
 
+        self.album_track_path, *_ = self.prepare_album_for_import(1)
+        self.single_path = self.prepare_track_for_import(2, self.import_path)
+        self.album_path = self.album_track_path.parent
+
     def tearDown(self):
-        self.teardown_beets()
+        super().tearDown()
         self.matcher.restore()
 
-    def __create_import_dir(self):
-        self._create_import_dir(1)
-        resource_path = os.path.join(_common.RSRC, b"empty.mp3")
-        single_path = os.path.join(self.import_dir, b"track_2.mp3")
-        shutil.copy(syspath(resource_path), syspath(single_path))
-        self.import_paths = [
-            os.path.join(self.import_dir, b"the_album"),
-            single_path,
-        ]
-        self.import_files = [
-            displayable_path(
-                os.path.join(self.import_paths[0], b"track_1.mp3")
-            ),
-            displayable_path(single_path),
-        ]
-
-    def __create_empty_import_dir(self):
-        path = os.path.join(self.temp_dir, b"empty")
-        os.makedirs(syspath(path))
-        self.empty_path = path
-
-    def __run(self, import_paths, singletons=True):
-        self._setup_import_session(singletons=singletons)
-        self.importer.paths = import_paths
-
+    def __run(self, importer):
         with capture_log() as logs:
-            self.importer.run()
-
-        logs = [line for line in logs if not line.startswith("Sending event:")]
+            importer.run()
 
         self.assertEqual(len(self.lib.items()), 0)
         self.assertEqual(len(self.lib.albums()), 0)
 
-        return logs
+        return {line for line in logs if not line.startswith("Sending event:")}
```

Comment:
Why have you made the log list a set?  Doesn't that obscure the order of the logs and whether they were produced multiple times?

Reply:
I don't think order should matter in this specific part, but the duplicate entry issue is something we should test for.

Reply:
Since I made the import dir to use the import directory as it is, it didn't seem possible to predict whether the album or the single gets picked up first (sometimes it's former and sometimes it's the latter).

Reply:
@Serene-Arc I think we could assert the number of imported items/albums to verify that duplicate imports are not happening. See `test_filefilter` where I replaced the logs assertions by library assertions.

Reply:
That would be a good solution I think, it would also skip the unlikely situation where the logs are wrong, which would be a separate issue but it's better to make sure that there's actually no duplicate imports.

Reply:
> Since I made the import dir to use the import directory as it is, it didn't seem possible to predict whether the album or the single gets picked up first (sometimes it's former and sometimes it's the latter).

I have reviewed this and found I am telling lies here: the import sequence is predictable - the importer picks up whichever is first in the alphabet, in this case album.

I have now made this a list.